src/smartpeak/source/ui/Widget.cpp

```diff
@@ -144,6 +144,12 @@ namespace SmartPeak
     table_scanned_ = (table_data_.body_.dimension(0) == table_entries_.size() && !data_changed_);
 
     if (table_data_.body_.dimensions().TotalSize() > 0) {
+      if (100 < table_data_.body_.dimension(0)) {
```

Comment:
Would be nice to have a named constant (here or at the beginning of the file) that you can reuse here instead of repeating 3 times "100".